transformer/src/getMethods.ts

```diff
@@ -1,13 +1,96 @@
-import * as ts                     from "typescript";
-import { Context }                 from "./contexts/Context";
-import { MethodDescriptionSource } from "./declarations";
-import { getDecorators }           from "./getDecorators";
-import { getTypeCall }             from "./getTypeCall";
+import * as ts                from "typescript";
+import { Context }            from "./contexts/Context";
+import {
+	MethodDescriptionSource,
+	ParameterDescriptionSource
+}                             from "./declarations";
+import { getDecorators }      from "./getDecorators";
+import { getTypeCall }        from "./getTypeCall";
 import {
 	getAccessModifier,
 	getFunctionLikeSignature,
-	getUnknownTypeCall
-}                                  from "./helpers";
+	getType,
+	getUnknownTypeCall,
+}                             from "./helpers";
+import {
+	color,
+	log,
+	LogLevel
+}                             from "./log";
+
+/**
+ * Process the signature of the method and create a parameter description for each parameter
+ *
+ * @param {ts.Signature} signature
+ * @param {Context} context
+ * @returns {Array<ParameterDescriptionSource>}
+ */
+export function getMethodParameters(signature: ts.Signature, context: Context): Array<ParameterDescriptionSource>
+{
+	const methodParameters = signature?.getParameters();
+
+	if (!signature || !methodParameters?.length)
+	{
+		return [];
+	}
+
+	const parameters: Array<ParameterDescriptionSource> = [];
+
+	for (let i = 0; i < methodParameters.length; i++)
+	{
+		const parameter = methodParameters[i];
+		const declaration = parameter.valueDeclaration as ts.ParameterDeclaration;
+		const type = getType(parameter, context.typeChecker);
+
+		if (declaration.type === undefined)
+		{
+			log.log(
+				LogLevel.Warning,
+				color.yellow,
+				`Failed to get type for parameter of method: ${signature.getDeclaration().name?.getText()}, parameter: ${parameter.name}`
+			);
+			continue;
+		}
+
+		// TODO: Figure why `methodParamsType({ name }: MethodParamsType)` {name} param is "__0" and type cannot be set to Type instance
+
+		const paramType = type && getTypeCall(type, type.symbol, context) || getUnknownTypeCall(context)
+
+		parameters.push({
+			n: parameter.getName(),
+			o: (parameter.flags & ts.TypeFlags.Object) === ts.TypeFlags.Object,
+			t: paramType,
+			i: i
```

Comment:
Is that `i` index for something special? It is an array of parameters so it is not difficult to find out in user code when needed..

Reply:
This is just so you have direct access to The index of the parameter you're handling

We can potentially skip a parameter being processed in the loop, so although you can check that array, it's not definitive if you're going to be inserting correctly at x index when dynamically calling a method 

If this makes sense? :D

Reply:
I rely on order everywhere. 
No constructor parameters, generic types etc.. are skipped, it keeps its position in arrays so index inside description object is redundant. + extra saved bytes of generated metadata. BUT, we can add this information to Parameter inside runtime package while mapping description into final object.

BTW, constructors and methods extend from same base class (in runtime package).

PS: https://docs.microsoft.com/en-us/dotnet/api/system.reflection.parameterinfo.position

PS 2: Are we going to implement method overloads? I've implemented overloads in [constructors](https://github.com/Hookyns/ts-reflection/blob/main/transformer/src/getConstructors.ts), it is important for me, but I have no real usage for methods yet. It just would be nice to have it. 

If we'll implement that, should be all the overloads in 1D array between other methods or should it be some property inside method description? C# keeps it in 1D array, but overloads are real methods there. In TS it is just one method with multiple parameter configurations, all matching one final method... hope you understand me. 😆

Reply:
So i guess rather than me calling continue in my loop for this, I should add a some kind of "null" type(i've seen this somewhere in the codebase already), this way we can rely on the order of returned descriptions and that everything is there

I could for sure look into method overloads, i don't see why not, my personal preference would be 1D array and have the information available on the method description